### PR TITLE
Listeners tab and filter visibility

### PR DESCRIPTION
**Pull request by vitorwtavares**
## Description

- Makes listener tab always visible, regardless of the number of listeners in the selected flow node.
- Hides listeners type filter whenever the selected flow node is not an User task.
- Disables listeners type filter whenever the selected flow node is an User task but returns no listeners on the initial api call (corresponds to the default 'All listeners' filter value).
- Adds different empty panel messages corresponding to different scenarios as described in the [designs on Figma](https://www.figma.com/design/zfOVIMNthZneqFiyUhWTP2/Listeners?node-id=2057-26289&node-type=section&m=dev).
- Refactors resetting of tab selection upon selecting a different flow node or flow node instance.
- Refactors style regarding bottom panel division (it will now have the expanded width when the Listeners tab is selected regardless of the amount of listeners).

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

relates to #25387

### CODE REVIEW

File: operate/client/playwright.config.ts [hunk 1]
```diff
@@ -26,7 +26,12 @@ const getPort = () => {
 };
 
 const getBasePath = () => {
-  return IS_CI && !IS_E2E ? '' : '/operate/';
+  const isLocalVisualRegression =
+    !IS_CI && !IS_E2E && !IS_A11Y && !IS_SCREENSHOT_GENERATOR;
+
+  const path = (IS_CI && !IS_E2E) || isLocalVisualRegression ? '' : '/operate/';
+
+  return path;
```
Comment (vitorwtavares) [inline operate/client/playwright.config.ts:34]: This fixes the Visual regression tests breaking locally.

There's still improvements we can do later on to make this solution more elegant, but for now this means E2E and Visual regression tests should be working again both on CI and locally when this gets merged.

File: operate/client/package.json [hunk 2]
```diff
@@ -46,7 +46,7 @@
     "eslint": "eslint src/ e2e-playwright/ --max-warnings=0",
     "lint": "yarn ts-check && yarn eslint",
     "fix:prettier": "prettier --write \"**/*.ts\"",
-    "test:e2e": "cross-env IS_E2E=true PORT=8081 ZEEBE_GATEWAY_ADDRESS=localhost:26503 ZEEBE_REST_GATEWAY_ADDRESS=http://localhost:8080 yarn playwright e2e-playwright/tests",
+    "test:e2e": "cross-env IS_E2E=true PORT=8081 ZEEBE_GATEWAY_ADDRESS=localhost:26503 ZEEBE_REST_GATEWAY_ADDRESS=http://localhost:8081 yarn playwright e2e-playwright/tests",
```
Comment (vitorwtavares) [inline operate/client/package.json:49]: @pedesen For context: Not sure if this was caused by the changes in exporter or something, but 8080 actually doesn't have anything running on it as far as I can tell when running the e2e backend (`make start-e2e`), we used to run the "regular" backend concurrently with it, so maybe that's why 8080 was used. 

TL;DR: Changing the REST port for 8081 finally made the C8 JS SDK work again (i.e. for completing a user task).

File: operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/reader/ElasticsearchUserTaskReader.java
Comment (vitorwtavares) [inline operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/reader/ElasticsearchUserTaskReader.java:1]: @sdorokhova There are changes to 2 backend Java files in this PR that you either added or I pulled from one of your branches, I believe due to an error related to the exporter a while back when we synced trying to get this unblocked.

Could you please confirm if these changes should be pushed or if they should be discarded (in case they have been resolved elsewhere)?

File: operate/webapp/src/main/java/io/camunda/operate/webapp/opensearch/reader/OpensearchUserTaskReader.java
Comment (vitorwtavares) [inline operate/webapp/src/main/java/io/camunda/operate/webapp/opensearch/reader/OpensearchUserTaskReader.java:1]: @sdorokhova This is the other file with changes I referenced in the other comment.

File: operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/reader/ElasticsearchUserTaskReader.java
Comment (sdorokhova) [reply operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/reader/ElasticsearchUserTaskReader.java:1]: In my opinion we should merge this. Since `tasklist-task` index stores not only tasks themselves, but also process instances and variables (similar to `operate-list-view` index), we should narrow down the search instead of using `match all` query. For now this method is used only in test, so it's anyway save to merge it.

File: operate/webapp/src/main/java/io/camunda/operate/webapp/opensearch/reader/OpensearchUserTaskReader.java
Comment (sdorokhova) [reply operate/webapp/src/main/java/io/camunda/operate/webapp/opensearch/reader/OpensearchUserTaskReader.java:1]: Same here. Let's merge it. It's the same logic but for Opensearch.

File: operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/reader/ElasticsearchUserTaskReader.java
Comment (vitorwtavares) [reply operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/reader/ElasticsearchUserTaskReader.java:1]: I'll keep the changes in this PR then, thank you!

File: operate/client/package.json (hunk 2 above)
Comment (pedesen) [reply operate/client/package.json:49]: It looks like this was changed by accidents a few weeks ago: https://github.com/camunda/camunda/commit/8e8d1c3787c255350540042d555b5c7b191fcbc7. So 8081 should be the right port here.

File: operate/client/src/App/ProcessInstance/BottomPanel/VariablePanel/Listeners/index.tsx [hunk 3]
```diff
@@ -42,117 +42,149 @@ type SelectedItem = {
...
   const [selectedOption, setSelectedOption] =
     useState<FilterLabelMappingKeys>('All listeners');
 
+  const handleEmptyMessages = () => {
+    if (isUserTask) {
+      if (selectedOption === 'All listeners')
+        return 'This flow node has no execution listeners nor user task listeners';
+
+      if (selectedOption === 'Execution listeners')
+        return 'This flow node has no execution listeners';
```
[the remark this answers is not in the document]
Comment (vitorwtavares) [reply operate/client/src/App/ProcessInstance/BottomPanel/VariablePanel/Listeners/index.tsx:67]: You're right, we can just use the default on this one, I'll adjust it.

File: operate/client/e2e-playwright/tests/processInstanceListeners.spec.ts [hunk 4]
```diff
@@ -145,18 +147,16 @@ test.describe('Process Instance Listeners', () => {
...
       page.getByTestId('state-overlay-completedEndEvents'),
     ).toBeVisible();
-    await expect
-      .poll(async () => {
-        await processInstancePage.diagram.clickFlowNode('Service Task B');
-        await processInstancePage.diagram.moveCanvasHorizontally(-200);
-        return processInstancePage.listenersTabButton.isVisible();
-      })
-      .toBeTruthy();
+    await page.waitForTimeout(1000);
```
[the remark this answers is not in the document]
Comment (vitorwtavares) [reply operate/client/e2e-playwright/tests/processInstanceListeners.spec.ts:155]: Yes, there is a delay until the data is actually updated regarding completing the user task and also showing the listeners. I tried evaluating a bunch of different elements both on the diagram and on the instance history but none of them actually waited the required time and the test ended up failing, the timeout was the only thing that fixed it. I'm aware it's discouraged and it's prone to flakiness, so I'm open for suggestions if you have any ideas.

[the remark this answers is not in the document]
Comment (pedesen) [reply operate/client/e2e-playwright/tests/processInstanceListeners.spec.ts:155]: Okay. Let's keep the delay for now. I'm also out of ideas for a check without timeout.

**Review by pedesen — APPROVED** [review]

**Comment by vitorwtavares** [comment]
Blocker from March 10th: PR has been approved but we currently have problems on the E2E tests on the CI preventing the merge. @pedesen Looked a bit into it on Monday, but hasn't had time to look further into it yet. I will also try to find some time for it on Wednesday (12th), or if we are sure the E2E issues are the same on main and nothing to do with this PR, it might get merged anyway to avoid more delays.

**Comment by vitorwtavares** [comment]
Mar 12th: After looking a bit into the E2E tests on main, I did not get to a solution but did confirm the issue has nothing to do with this PR, hence I'm merging it into main.

I've [created an issue](https://github.com/camunda/camunda/issues/29527) and assigned myself and @pedesen to follow up on the E2E tests for Operate as soon as one of us have some time (need to align prioritization with EM's of the current projects we're working on).

**Comment by vitorwtavares** [comment]
@ArezooAzar Can you QA this? I've added the steps to do it on the [description of the issue](https://github.com/camunda/camunda/issues/25387). Feel free to message me if something's not clear.

**Comment by ArezooAzar** [comment]
@vitorwtavares : 


I tried to test the PR in 
- **SaaS:** Not working due to unhealthy Operate and Tasklist.  
- **SM:** I can deploy the process, but it doesn’t appear in Operate.  
- **Helm chart:** I created the environment, and everything is green, but the provided link is not working (I've already contacted the distro team).  

Testing is blocked due to the lack of a test environment.

**Comment by vitorwtavares** [comment]
> @vitorwtavares :
> 
> I tried to test the PR in
> 
> * **SaaS:** Not working due to unhealthy Operate and Tasklist.
> * **SM:** I can deploy the process, but it doesn’t appear in Operate.
> * **Helm chart:** I created the environment, and everything is green, but the provided link is not working (I've already contacted the distro team).
> 
> Testing is blocked due to the lack of a test environment.

@ArezooAzar I'm not sure I can help with any of those. Did you reach out to someone to figure out why these environments are not working (or is it a known issue?)?
